agent/src/main/java/com/teamscale/jacoco/agent/Agent.java

```diff
@@ -142,14 +149,19 @@ private void dumpReportUnsafe() {
 			return;
 		}
 
-		String xml;
+		File coverageFile;
+		long currentTime = System.currentTimeMillis();
+		Path outputFile = this.outputDirectory.resolve("jacoco-" + currentTime + ".xml");
 		try (Benchmark benchmark = new Benchmark("Generating the XML report")) {
-			xml = generator.convert(dump);
+			coverageFile = generator.convert(dump, outputFile.toString());
 		} catch (IOException e) {
 			logger.error("Converting binary dump to XML failed", e);
 			return;
 		}
 
-		store.store(xml);
+		if (coverageFile != null) {
```

Comment:
Could have happened when `ouputFile.createNewFile()` in line 76 of class `JaCoCoXmlReportGenerator` returned false. I read the documentation again, this would happen if the file already existed. So instead of returning null, I changed the code to clear the file contents instead. => we don't need this null check anymore :)